ols/app/models/config.py

```diff
@@ -308,11 +308,43 @@ def validate_yaml(self) -> None:
         """Validate logger config."""
 
 
+class ReferenceContent(BaseModel):
+    """Reference content configuration."""
+
+    product_docs_persist_dir: Optional[str] = constants.PRODUCT_DOCS_PERSIST_DIR
+    product_index: Optional[str] = constants.PRODUCT_INDEX
+
+    def __init__(self, data: Optional[dict] = None):
+        """Initialize configuration and perform basic validation."""
+        super().__init__()
+        if data is None:
+            data = {}
+
+        self.product_docs_persist_dir = data.get(
+            "product_docs_persist_dir", constants.PRODUCT_DOCS_PERSIST_DIR
+        )
+        self.product_index = data.get("product_index", constants.PRODUCT_INDEX)
+
+    def __eq__(self, other):
+        """Compare two objects for equality."""
+        if isinstance(other, ReferenceContent):
+            return (
+                self.product_docs_persist_dir == other.product_docs_persist_dir
+                and self.product_index == other.product_index
+            )
+        return False
+
+    def validate_yaml(self) -> None:
+        """Validate reference content config."""
+        pass
```

Comment:
`pass` isn't needed, evidently python treats the code comment as sufficient.

but also we should be validating that the value of product_docs_persist_dir is a valid filesystem path format.  Should probably even confirm a file exists at that location so we can fail on startup if it doesn't exist, rather than failing when a request comes in and we can't load the index.

Reply:
I thought about it, but then every one will start failing as they have been running without the index file. Let me know and I can fail if the directory or the files does not exist.

The other thing is that as per your previous comment to take the reference content configuration out of olsconfig i created a separate section in config called indexer_config where the configuration for indexer can go. Let me know your thoughts.

Reply:
> I thought about it, but then every one will start failing as they have been running without the index file. 

they won't fail if we default the value to None and tolerate None as valid.

>  i created a separate section in config called indexer_config where the configuration for indexer can go.

indexer config does not belong in the ols config.yaml at all.  The indexer is a standalone utility that no user is ever going to use.  I am going to move it out of this directory so it stops causing people to be confused.

Reply:
https://github.com/openshift/lightspeed-service/pull/299